horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -107,7 +108,128 @@ public void testTransformationWithoutSchema(TestInfo info) throws InterruptedExc
       assertFalse(run.trashed);
       assertNewDataset(dataSetQueue, runId);
    }
+   private int labelValues_setup(Test t) throws JsonProcessingException {
+      Schema fooSchema = createSchema("foo","urn:foo");
+      Extractor fooExtractor = new Extractor();
+      fooExtractor.name="foo";
+      fooExtractor.jsonpath="$.foo";
+      Extractor barExtractor = new Extractor();
+      barExtractor.name="bar";
+      barExtractor.jsonpath="$.bar";
+      addLabel(fooSchema,"labelFoo","",fooExtractor);
+      addLabel(fooSchema,"labelBar","",barExtractor);
+
+
+      ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);;
+      JsonNode data = mapper.readTree("{ \"foo\": \"uno\", \"bar\": \"dox\"}");
+      String idString = uploadRun(data,t.name,fooSchema.uri);
+      int id = Integer.parseInt(idString);
+      return id;
+   }
+
+   @org.junit.jupiter.api.Test
+   public void labelValues_include_excluded() throws JsonProcessingException {
+      Test t = createTest(createExampleTest("my-test"));
+      int id = labelValues_setup(t);
+
+      JsonNode response = jsonRequest()
+              .get("/api/run/"+id+"/labelValues?include=labelFoo&exclude=labelFoo")
+              .then()
+              .statusCode(200)
+              .extract()
+              .body()
+              .as(JsonNode.class);
+      assertInstanceOf(ArrayNode.class,response);
+      ArrayNode arrayResponse = (ArrayNode)response;
+      assertEquals(1,arrayResponse.size());
+      assertInstanceOf(ObjectNode.class,arrayResponse.get(0));
+      ObjectNode objectNode = (ObjectNode)arrayResponse.get(0).get("values");
+      assertFalse(objectNode.has("labelFoo"),objectNode.toString());
+      assertTrue(objectNode.has("labelBar"),objectNode.toString());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void labelValues_include_two_params() throws JsonProcessingException {
+      Test t = createTest(createExampleTest("my-test"));
+      int id = labelValues_setup(t);
+
+      JsonNode response = jsonRequest()
+              .get("/api/run/"+id+"/labelValues?include=labelFoo&include=labelBar")
+              .then()
+              .statusCode(200)
+              .extract()
+              .body()
+              .as(JsonNode.class);
+      assertInstanceOf(ArrayNode.class,response);
+      ArrayNode arrayResponse = (ArrayNode)response;
+      assertEquals(1,arrayResponse.size());
+      assertInstanceOf(ObjectNode.class,arrayResponse.get(0));
+      ObjectNode objectNode = (ObjectNode)arrayResponse.get(0).get("values");
+      assertTrue(objectNode.has("labelFoo"),objectNode.toString());
+      assertTrue(objectNode.has("labelBar"),objectNode.toString());
+   }
+   @org.junit.jupiter.api.Test
+   public void labelValues_include_two_separated() throws JsonProcessingException {
```

Comment:
rename to labelValuesIncludeTwoSeparated?

Reply:
renamed